src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -163,7 +163,10 @@ def run_mode()
   def terminate
     Puppet::Server::Config.terminate_puppet_server
     # See https://github.com/jruby/jruby/issues/7349
-    Timeout.instance_variable_get(:@timeout_thread).exit
+    timeout_thread = Timeout.instance_variable_get(:@timeout_thread)
```

Comment:
```suggestion
    Timeout.instance_variable_get(:@timeout_thread)&.exit
```
Maybe safe operator?

Reply:
kids these days and their operators